iroh/examples/0rtt.rs

```diff
@@ -106,10 +71,36 @@ async fn connect(args: Args) -> n0_snafu::Result<()> {
         let connection = if args.disable_0rtt {
             let connection = connecting.await.e()?;
             trace!("connecting without 0-RTT");
-            pingpong(&connection, future::ready(true), i).await?;
+            let (send, recv) = connection.open_bi().await.e()?;
+            pingpong(send, recv, i).await?;
             connection
         } else {
-            pingpong_0rtt(connecting, i).await?
+            match connecting.into_0rtt() {
+                Ok(zrtt_connection) => {
+                    trace!("0-RTT possible from our side");
+                    let (send, recv) = zrtt_connection.open_bi().await.e()?;
+                    let zrtt_task = tokio::spawn(pingpong(send, recv, i));
+                    match zrtt_connection.handshake_completed().await {
+                        ZeroRtt::Accepted(conn) => {
+                            let _ = zrtt_task.await.e()?;
+                            conn
+                        }
+                        ZeroRtt::Rejected(conn) => {
+                            zrtt_task.abort();
+                            let (send, recv) = conn.open_bi().await.e()?;
+                            pingpong(send, recv, i).await?;
+                            conn
+                        }
```

Comment:
Can we revert these changes to the example?
I personally find this version harder to understand:
1. We introduce more parallelism via a `tokio::spawn` that we didn't have before
2. Before, we checked whether our "ping sending" went through, and if it didn't we send it again after the handshake completed. But now, we also duplicate the `recv.read_to_end`. I don't know what that means for spawned task. Does that just hang another recv future?

I do really think I'd prefer the version where we check the "proceed" future in the `pingpong` function.

Perhaps I'm missing something though, I'm sure you've changed it because you found the old version confusing. Perhaps we can find a better third version if you explain what was confusing about the original example code?